### PR TITLE
Prevent dupe participation in same experiment

### PR DESCRIPTION
**Pull request by pihman**
There's a race condition that can lead to duplicate participation -- and possibly conflicting segmentation -- for a given experiment.

Thread A:   `Vanity.ab_test(:my_experiment)`

Thread B:   `Vanity.ab_test(:my_experiment)`

---

To prevent this situation:

1) Make the existing (experiment_id, identity) index a unique index.
2) Update the Vanity::Participant code to gracefully handle the dup record failure.

### CODE REVIEW

**Comment by phillbaker** [comment]
Thanks for the PR, it'll take me a day or two to review.

File: lib/generators/templates/add_unique_indexes_migration.rb [hunk 1]
```diff
@@ -17,11 +17,8 @@ def with_vanity_connection
 
   def change
     with_vanity_connection do
-      remove_index :vanity_experiments, [:experiment_id]
```
Comment (phillbaker) [inline lib/generators/templates/add_unique_indexes_migration.rb]: I'm a bit confused - why did you move these indexes into `add_participants_unique_index_migration.rb` and add the participants index here? Did you mean to do the opposite?

Comment (pihman) [reply lib/generators/templates/add_unique_indexes_migration.rb]: ah yes, the additions should have been made to a separate file `add_participants_unique_index_migration.rb` instead of replacing the pre-existing migration.  let me fix that.

**Comment by pihman** [comment]
@phillbaker I un-mixed up the migrations.  Thanks!
